File: 1.Domain/Domain.Model/Employee.cs
using Core;
using Domain.Model.Enum;
using System;
using System.Collections.Generic;
using System.Text;

namespace Domain.Model
{
    public class Employee : Entity<int>
    {
        public string Name { get; set; }
        public string LastName { get; set; }
        public int DNI { get; set; }
        public string PhoneNumber { get; set; }
        public string EmailAddress { get; set; }
        public string LinkedInProfile { get; set; }
        public string AdditionalInformation { get; set; }
        public EmployeeStatus Status { get; set; }
        public Consultant Recruiter { get; set; }
        public Role Role { get; set; }
        public bool isReviewer { get; set; }
        public Employee Reviewer { get; set; }
    }
}

File: 4.Clients/WebApi/ClientApp/src/entities/employee.ts
import { Role } from "./role";

export class Employee {
    constructor
        (public id: number, public name: string,
            public lastName: string, public phoneNumber: string,
            public emailAddress: string, public linkedInProfile: string,
            public additionalInformation: string, public status: number,
            public recruiterId: number, public dni: number,
            public role: Role, public roleId: number,
            public isReviewer: boolean, public reviewer: Employee, public reviewerId: number) { }
}